jobs/submanPipeline.groovy

```diff
@@ -0,0 +1,79 @@
+String REPO = 'https://github.com/candlepin/subscription-manager'
+
+def setCommitStatus = { repo, sha1, state, message ->
+    try {
+        step([
+            $class: 'GitHubCommitStatusSetter',
+            commitShaSource: [
+                $class: 'ManuallyEnteredShaSource',
+                sha   : sha1,
+            ],
+            contextSource: [
+                $class: 'ManuallyEnteredCommitContextSource',
+                context: 'jenkins-pipeline',
+            ],
+            reposSource: [
+                $class: 'ManuallyEnteredRepositorySource',
+                url   : repo,
+            ],
+            statusResultSource: [
+                $class : 'ConditionalStatusResultSource',
+                results: [[
+                    $class : 'AnyBuildResult',
+                    message: message,
+                    state  : state,
+                ]]
+            ],
+        ])
+    } catch(e) {
+        echo "Unable to set status for ${sha1}: ${e}"
+    }
+}
+
+stage('test') {
+    node('master') {
+        setCommitStatus(REPO, "${sha1}", 'Pending', 'Tests are pending.')
+    }
+    try {
+        parallel(
+            'nose': {
+                build(job: 'subscription-manager-nose-tests-pr', parameters: [[
+                    $class: 'StringParameterValue',
+                    name: 'sha1',
+                    value: "${sha1}"
+                ]])
+            },
+            'stylish': {
+                build(job: 'subscription-manager-stylish-tests-pr', parameters: [[
+                    $class: 'StringParameterValue',
+                    name: 'sha1',
+                    value: "${sha1}"
+                ]])
+            },
+            'tito': {
+                build(job: 'subscription-manager-tito-tests-pr', parameters: [[
+                    $class: 'StringParameterValue',
+                    name: 'sha1',
+                    value: "${sha1}"
+                ]])
+            },
+        )
+        node('master') {
+            currentBuild.result = 'SUCCESS'
+            setCommitStatus(REPO, "${sha1}", 'SUCCESS', 'All tests passed.')
+        }
+    } catch(e) {
+        node('master') {
+            currentBuild.result = 'FAILURE'
+            setCommitStatus(REPO, "${sha1}", 'FAILURE', 'One or more tests failed.')
+
+            String subject = "subscription-manager - build # ${env.BUILD_NUMBER} - Failure"
+            emailext(
+                    subject: subject,
+                    body: "${subject}:\n\nCheck console output at ${env.BUILD_URL} to view the results",
+                    to: 'chainsaw@redhat.com',
+            )
+            // TODO figure out how to do IRC notification
```

Comment:
A few options here:
1) this ugly shell script: https://jenkins.io/doc/pipeline/examples/#ircnotify-commandline
2) we could wrap the whole pipeline in the helper job (i.e. have the helper job block waiting on the pipeline to finish, and then have *it* do the IRC notify based on the pipeline result. Downside here is that it would occupy an extra executor.
3) Access the Jenkins IRC code from inside the pipeline script.
4) Hack on the IRC plugin?

Open to other ideas. @Lorquas @barnabycourt @Ceiu ?